experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt

```diff
@@ -5,50 +5,38 @@ import com.automattic.android.experimentation.ExPlat.RefreshStrategy.IF_STALE
 import com.automattic.android.experimentation.ExPlat.RefreshStrategy.NEVER
 import com.automattic.android.experimentation.domain.Assignments
 import com.automattic.android.experimentation.domain.AssignmentsValidator
-import com.automattic.android.experimentation.domain.SystemClock
 import com.automattic.android.experimentation.domain.Variation
 import com.automattic.android.experimentation.domain.Variation.Control
-import com.automattic.android.experimentation.local.FileBasedCache
-import com.automattic.android.experimentation.remote.ExperimentRestClient
 import com.automattic.android.experimentation.repository.AssignmentsRepository
-import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import java.io.File
 
-class ExPlat internal constructor(
+public class ExPlat internal constructor(
     private val platform: String,
-    private val experiments: Set<Experiment>,
+    experiments: Set<Experiment>,
     private val logger: ExperimentLogger,
     private val coroutineScope: CoroutineScope,
     private val isDebug: Boolean,
     private val assignmentsValidator: AssignmentsValidator,
     private val repository: AssignmentsRepository,
-) {
+) : VariationsRepository {
     private val activeVariations = mutableMapOf<String, Variation>()
     private val experimentIdentifiers: List<String> = experiments.map { it.identifier }
 
     private var anonId: String? = null
 
-    fun configure(anonId: String? = null) {
+    init {
+        coroutineScope.launch {
+            refresh()
```

Comment:
Question (❓): I think this is a good move, but how about when users would like to also use `configure(...)` to setup ExPlat with an anonymous id, they will then need to trigger `refresh()` manually once more, right, or, how will they better use the API then? 🤔

Reply:
True, this was not the best idea. I learned that `anonId` is actually a mandatory information ( p1725007206945009-slack-C7HH3V5AS) and I proposed changes to this initialization methods in #245

Reply:
Thanks for all this work @wzieba , I followed all the conversation on anonymous id needing to become mandatory (and all), plus did a quick review on #245 , the changes makes sense to me (also, thanks @iangmaia for the review)! 🚀